### ISSUE TITLE
MacOS Sierra: address exclusion doesn't work.

### ISSUE THREAD

**Issue by mr-bat** [sshuttle/sshuttle#164]
I have a local server running on port 4200; whenever I invoked 'sshuttle', I would lose my connection to all my local servers and had access to the localhost of the server. So then, I excluded localhost in the forward list via `sshuttle --dns -x 127.0.0.1 -vr my_server 0.0.0.0/0`. To my surprise it did nothing and still, my traffic to localhost was being forwarded. I tried different combinations and at last decided not to forward any traffic being made to any IP on port 4200 (`sshuttle -vr my_server 0.0.0.0/0 -x 0.0.0.0/0:4200`). But in my surprise, even this lazy command didn't make it and I believe there is something wrong with the exclusion part as it always sends **DNS** queries to remote server.
sshuttle -V: `0.78.2.dev27+gf9361d7`
cat /etc/resolv.conf:
```#
# Mac OS X Notice
#
# This file is not used by the host name and address resolution
# or the DNS query routing mechanisms used by most processes on
# this Mac OS X system.
#
# This file is automatically generated.
#
nameserver 192.168.1.1
```

**Comment by mr-bat** [comment]
`sshuttle -vr my_server -x 0.0.0.0/0:4200 0.0.0.0/0`:
```
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 2.7.10
firewall manager: ready method name pf.
IPv6 enabled: True
UDP enabled: False
DNS enabled: False
TCP redirector listening on ('::1', 12300, 0, 0).
TCP redirector listening on ('127.0.0.1', 12300).
Starting client with Python version 2.7.10
c : connecting to server...
Starting server with Python version 3.5.2
 s: latency control setting = True
 s: available routes:
 s:   2/10.8.0.0/24
 s:   2/10.18.0.0/16
 s:   2/178.62.192.0/18
c : Connected.
firewall manager: setting up.
>> pfctl -s Interfaces -i lo -v
>> pfctl -s all
>> pfctl -a sshuttle6-12300 -f /dev/stdin
>> pfctl -E
>> pfctl -s Interfaces -i lo -v
>> pfctl -s all
>> pfctl -a sshuttle-12300 -f /dev/stdin
>> pfctl -E
c : Accept TCP: 127.0.0.1:58438 -> 127.0.0.1:4200.
 s: SW#6:127.0.0.1:4200: uwrite: got EPIPE
c : Accept TCP: 127.0.0.1:58439 -> 127.0.0.1:4200.
c : SW#9:127.0.0.1:58438: deleting (3 remain)
c : SW'unknown':Mux#1: deleting (2 remain)
 s: SW#7:127.0.0.1:4200: uwrite: got EPIPE
c : Accept TCP: 127.0.0.1:58440 -> 127.0.0.1:4200.
c : SW#12:127.0.0.1:58439: deleting (3 remain)
c : SW'unknown':Mux#2: deleting (2 remain)
 s: warning: closed channel 3 got cmd=TCP_DATA len=361
c : Accept TCP: 127.0.0.1:58441 -> 127.0.0.1:4200.
c : SW#9:127.0.0.1:58440: deleting (3 remain)
c : SW'unknown':Mux#3: deleting (2 remain)
 s: warning: closed channel 4 got cmd=TCP_DATA len=361
c : Accept TCP: 127.0.0.1:58442 -> 127.0.0.1:4200.
c : SW#12:127.0.0.1:58441: deleting (3 remain)
c : SW'unknown':Mux#4: deleting (2 remain)
 s: warning: closed channel 5 got cmd=TCP_DATA len=361
c : Accept TCP: 127.0.0.1:58443 -> 127.0.0.1:4200.
c : SW#9:127.0.0.1:58442: deleting (3 remain)
c : SW'unknown':Mux#5: deleting (2 remain)
 s: warning: closed channel 6 got cmd=TCP_DATA len=361
c : Accept TCP: 127.0.0.1:58444 -> 127.0.0.1:4200.
c : SW#12:127.0.0.1:58443: deleting (3 remain)
c : SW'unknown':Mux#6: deleting (2 remain)
 s: warning: closed channel 7 got cmd=TCP_DATA len=361
c : Accept TCP: 127.0.0.1:58445 -> 127.0.0.1:4200.
c : SW#9:127.0.0.1:58444: deleting (3 remain)
c : SW'unknown':Mux#7: deleting (2 remain)
c : Accept TCP: 127.0.0.1:58446 -> 127.0.0.1:4200.
c : SW#12:127.0.0.1:58445: deleting (3 remain)
c : SW'unknown':Mux#8: deleting (2 remain)
 s: warning: closed channel 8 got cmd=TCP_DATA len=361
 s: SW#14:127.0.0.1:4200: uwrite: got EPIPE
c : Accept TCP: 127.0.0.1:58447 -> 127.0.0.1:4200.
c : SW#9:127.0.0.1:58446: deleting (3 remain)
c : SW'unknown':Mux#9: deleting (2 remain)
 s: warning: closed channel 10 got cmd=TCP_DATA len=361
```

**Comment by vieira** [comment]
Hello @mr-bat 

This is probably related to #159 and should be fixed in #162. Can you test?

**Comment by mr-bat** [comment]
Actually, I built the program via the latest commit so it means your PR is also included.

**Comment by vieira** [comment]
@mr-bat Do you have any other version of sshuttle installed? Please post your `sshuttle --version` and the output with `-vv` instead of just `-v` please.

If you have other older version of sshuttle installed it might be executing instead of the one you built from git HEAD.

**Comment by mr-bat** [comment]
`sshuttle --version/ sshuttle --version -v/ sshuttle --version -vv`
They all lead to number `0.78.1`.

**Comment by vieira** [comment]
Hello @mr-bat,

That is not HEAD (and not even the latest version which is 0.78.3). You probably have some version of sshuttle installed that is getting picked instead of the version you cloned. 

If you used `pip` or `brew` to install you can use that same tool to uninstall and then run the version you cloned with `./run`.

Thanks.

**Comment by mr-bat** [comment]
Thank you. While I solved my problem via the solution in [#159](https://github.com/sshuttle/sshuttle/issues/159) but also upgrading to 0.78.4 also solved my problem.
